File: android/src/main/java/com/reactnativecompressor/Image/utils/ImageCompressorOptions.java
package com.reactnativecompressor.Image.utils;

import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.ReadableMapKeySetIterator;

public class ImageCompressorOptions {
    public static ImageCompressorOptions fromMap(ReadableMap map) {
        final ImageCompressorOptions options = new ImageCompressorOptions();
        final ReadableMapKeySetIterator iterator = map.keySetIterator();

        while (iterator.hasNextKey()) {
            final String key = iterator.nextKey();

            switch (key) {
                case "compressionMethod":
                     options.compressionMethod = CompressionMethod.valueOf(map.getString(key));
                      break;
                case "maxWidth":
                    options.maxWidth = map.getInt(key);
                    break;
                case "maxHeight":
                    options.maxHeight = map.getInt(key);
                    break;
                case "quality":
                    options.quality = (float) map.getDouble(key);
                    break;
                case "input":
                    options.input = InputType.valueOf(map.getString(key));
                    break;
                case "output":
                    options.output = OutputType.valueOf(map.getString(key));
                    break;
              case "returnableOutputType":
                options.returnableOutputType = ReturnableOutputType.valueOf(map.getString(key));
                break;
            }
        }

        return options;
    }

    public enum InputType {
        base64, uri
    }

    public enum OutputType {
        png, jpg
    }

  public enum ReturnableOutputType {
    base64, uri
  }

  public enum CompressionMethod {
    auto, manual
  }

    public CompressionMethod compressionMethod = CompressionMethod.manual;
    public int maxWidth = 1280;
    public int maxHeight = 1280;
    public float quality = 0.8f;
    public InputType input = InputType.uri;
    public OutputType output = OutputType.jpg;
    public ReturnableOutputType returnableOutputType = ReturnableOutputType.uri;
}

File: android/src/main/java/com/reactnativecompressor/Image/ImageCompressorOptions.kt
package com.reactnativecompressor.Image

import com.facebook.react.bridge.ReadableMap

class ImageCompressorOptions {
    enum class InputType {
        base64,
        uri
    }

    enum class OutputType {
        png,
        jpg
    }

    enum class ReturnableOutputType {
        base64,
        uri
    }

    enum class CompressionMethod {
        auto,
        manual
    }

    var compressionMethod = CompressionMethod.auto
    var maxWidth = 1280
    var maxHeight = 1280
    var progressDivider: Int? = 0
    var quality = 0.8f
    var input = InputType.uri
    var output = OutputType.jpg
    var uuid: String? = ""
    var returnableOutputType = ReturnableOutputType.uri
    var disablePngTransparency:Boolean = false

    companion object {
        fun fromMap(map: ReadableMap): ImageCompressorOptions {
            val options = ImageCompressorOptions()
            val iterator = map.keySetIterator()
            while (iterator.hasNextKey()) {
                val key = iterator.nextKey()
                when (key) {
                    "compressionMethod" -> options.compressionMethod = CompressionMethod.valueOf(map.getString(key)!!)
                    "maxWidth" -> options.maxWidth = map.getInt(key)
                    "maxHeight" -> options.maxHeight = map.getInt(key)
                    "progressDivider" -> options.progressDivider = map.getInt(key)
                    "quality" -> options.quality = map.getDouble(key).toFloat()
                    "input" -> options.input = InputType.valueOf(map.getString(key)!!)
                    "output" -> options.output = OutputType.valueOf(map.getString(key)!!)
                    "returnableOutputType" -> options.returnableOutputType = ReturnableOutputType.valueOf(map.getString(key)!!)
                    "uuid" -> options.uuid = map.getString(key)
                    "disablePngTransparency" -> options.disablePngTransparency = map.getBoolean(key)
                }
            }
            return options
        }
    }
}
